src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/OasysService.kt

```diff
@@ -122,6 +125,20 @@ class OasysService(val oasysApiClient: OasysApiClient) {
     return oasysPsychiatric.toModel()
   }
 
+  fun getLearningNeeds(prisonNumber: String): LearningNeeds? {
+    val assessmentId = getAssessmentId(prisonNumber)
```

Comment:
Can we do something like this? Thoughts

We can get rid of the if else check

val assessmentId = getAssessmentId(prisonNumber)
        ?: throw NotFoundException("No learning needs information found for prison number $prisonNumber")

Reply:
yes good suggestion!